src/Operation/FindOneAndReplace.php

```diff
@@ -147,6 +158,15 @@ public function __construct(string $databaseName, string $collectionName, $filte
 
         unset($options['projection'], $options['returnDocument']);
 
+        if (isset($options['codec'])) {
+            $replacement = $options['codec']->encodeIfSupported($replacement);
+        }
+
+        // Psalm's assert-if-true annotation does not work with unions, so
+        // assert the type manually instead of using is_document
+        // See https://github.com/vimeo/psalm/issues/6831
+        assert(is_array($replacement) || is_object($replacement));
```

Comment:
We already do `if (! is_document($replacement))` earlier in this method, so I think this assertion can be moved within `if (isset($options['codec']))`. I assume it's only needed to assert the result of `encodeIfSupported()`, correct?

I think that was in my original suggestion but overlooked.

Reply:
Good point. Changed 👍